tests/components/shelly/test_sensor.py

```diff
@@ -153,7 +159,11 @@ async def test_block_restored_sleeping_sensor(
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
 
-    assert hass.states.get(entity_id).state == "20.4"
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == "20.4"
+    assert state.attributes[ATTR_STATE_CLASS] == SensorStateClass.MEASUREMENT
+    assert state.attributes[ATTR_DEVICE_CLASS] == SensorDeviceClass.TEMPERATURE
```

Comment:
This was added to verify that the change fix the issue, test will fail if run without the changes in this PR
```log
   FAILED tests/components/shelly/test_sensor.py::test_block_restored_sleeping_sensor - AssertionError: assert None == <SensorStateClass.MEASUREMENT: 'measurement'>
   FAILED tests/components/shelly/test_sensor.py::test_block_restored_sleeping_sensor - AssertionError: assert None == <SensorDeviceClass.TEMPERATURE: 'temperature'>
```